cupy/cuda/memory.pyx

```diff
@@ -441,7 +442,38 @@ cdef class SingleDeviceMemoryPool:
             merged.next.prev = merged
         return merged
 
+    cpdef MemoryPointer _alloc(self, Py_ssize_t rounded_size):
+        hooks = get_memory_hooks()
+        if hooks:
+            hooks_values = hooks.values()
```

Comment:
I looked cython, and it seemed cython's `OrderdDict` does not have `itervalues()` unlike `defaultdict`. https://github.com/cupy/cupy/pull/274.
So, I fixed to use six for this.